### PR TITLE
Update RSPEC before 9.8 release

### LINKED ISSUE #7697: Update RSPEC before 9.8 release
cc @tom-howlett-sonarsource
RSPEC changes:
* https://github.com/SonarSource/rspec/pull/2716 From Taxonomy Classification sprint: was supposed to target [9.7](https://github.com/SonarSource/sonar-dotnet/issues/7645), but, while approved, wasn't merged in time due to a [validator issue](https://cirrus-ci.com/task/5416645879922688) on the RSPEC CI.
* https://github.com/SonarSource/rspec/pull/2827 Originated from [this dogfood post](https://discuss.sonarsource.com/t/wrong-diff-tags-in-rules-descriptions/15367).
* Fixes in S2589 and S2583 for VB.NET (https://github.com/SonarSource/rspec/pull/2856 and https://github.com/SonarSource/rspec/pull/2857)
* Removes clean code taxonomy tag for new VB.NET rules S2589 and S2583. Tags will be taken care of in the next release.

### PR DESCRIPTION
**Pull request by mary-georgiou-sonarsource**
Fixes #7697

### CODE REVIEW

File: analyzers/rspec/vbnet/S2589.json [hunk 1]
```diff
@@ -1,12 +1,6 @@
 {
   "title": "Boolean expressions should not be gratuitous",
   "type": "CODE_SMELL",
-  "code": {
```
Comment (mary-georgiou-sonarsource) [inline analyzers/rspec/vbnet/S2589.json:4]: Removing CCT tags - to be take care of in the next release.

File: analyzers/rspec/vbnet/S2583.json [hunk 2]
```diff
@@ -1,12 +1,6 @@
 {
   "title": "Conditionally executed code should be reachable",
   "type": "BUG",
-  "code": {
```
Comment (mary-georgiou-sonarsource) [inline analyzers/rspec/vbnet/S2583.json:4]: Removing  CCT tag - to be taken care of in the next release.

**Review by csaba-sagi-sonarsource — APPROVED** [review]
LGTM!

**Review by Tim-Pohlmann — APPROVED** [review]
LGTM

**Pull request merged**